File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        def traverse(root,level):
            nonlocal res
            if not root: return
            if (len(res) <= level): res.append([])
            res[level].append(root.val)
            traverse(root.left,level+1)
            traverse(root.right,level+1)
        traverse(root,0)
        result = [level[-1] for level in res]
        return result

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
      void traverseToVec(TreeNode *node, vector<int> &res) {
    if (!node) {
      return;
    }
    traverseToVec(node->left, res);
    res.push_back(node->val);
    traverseToVec(node->right, res);
  };
  int kthSmallest(TreeNode *root, int k) {
    if (!root)
      return 0;
    vector<int> res;
    traverseToVec(root, res);
    return res[k-1];
  }

};
